repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -122,7 +122,11 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
     }
 
     val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable match {
+      case x if x.nonEmpty => x
```

Comment:
Also, I realize you might be using the guard so that you can bind x. Of course, you could always return `myCollection`.

In the end, it might be a preference thing... I just don't really like unnecessary guards. Thinking about it, though, it may actually be more efficient in this case.